src/flag_gems/ops/masked_fill.py

```diff
@@ -0,0 +1,64 @@
+import logging
+
+import torch
+import triton
+import triton.language as tl
+
+from ..utils import broadcastable_to, libentry
+
+
+def cfggen():
+    block_m = [1, 2, 4, 8]
+    configs = [
+        triton.Config({"BLOCK_M": m, "BLOCK_N": 128}, num_warps=4) for m in block_m
+    ]
+    return configs
+
+
+@libentry()
+@triton.autotune(configs=cfggen(), key=["M", "N"])
+@triton.jit
+def masked_fill_kernel(
+    inp, expand_mask, value, out, M, N, BLOCK_M: tl.constexpr, BLOCK_N: tl.constexpr
+):
+    pid = tl.program_id(0)
+    rows_offset = pid * BLOCK_M + tl.arange(0, BLOCK_M)[:, None]
+    rows_mask = rows_offset < M
+
+    for off in range(0, N, BLOCK_N):
+        cols_offset = off + tl.arange(0, BLOCK_N)[None, :]
+        cols_mask = cols_offset < N
+        block_mask = rows_mask and cols_mask
+
+        offsets = rows_offset * N + cols_offset

```

Comment:
This assumes c-contiguous input and 
```
tl.load(expand_mask + offsets...
tl.load(expand_mask + offsets...
tl.load(inp + offsets...
```

Reply:
Thanks, it definitely needs thinking about. Already updated the relevant code.